src/states/elder/mod.rs

```diff
@@ -381,18 +381,19 @@ impl Elder {
     }
 
     fn reset_parsec_with_data(&mut self, reset_data: ParsecResetData) -> Result<(), RoutingError> {
-        let drained_obs: Vec<_> = self
-            .parsec_map
-            .our_unpolled_observations()
-            .cloned()
-            .collect();
-
         let ParsecResetData {
             gen_pfx_info,
             mut cached_events,
             completed_events,
         } = reset_data;
+
         self.gen_pfx_info = gen_pfx_info;
+
+        let drained_obs: Vec<_> = self
```

Comment:
Any reason for this change?

Reply:
moved position when resolve conflicts due to rebase. Will change it back.